File: app/src/main/java/com/matthewcannefax/learnsightwords/word/WordViewModel.java
package com.matthewcannefax.learnsightwords.word;

import android.app.Application;
import android.content.Context;

import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import com.matthewcannefax.learnsightwords.room.WordRepo;
import com.matthewcannefax.learnsightwords.speech.SpeechHelper;
import com.matthewcannefax.learnsightwords.word.Word;

import java.util.List;

public class WordViewModel extends AndroidViewModel {

    //reference the database repository
    private WordRepo mRepo;

    //mutable live object to watch which word is the current word
    //the text view will change when the currentSightWord changes
    public MutableLiveData<Word> currentSightWord;

    //just a typical list from the database
    //used to manage the currentSightWord
    private List<Word> wordList;

    //also used to manage the currentSightWord
    private int currentIndex;

    //application context
    private Context context;

    //not really using this currently
    //might try to use in the future instead of wordList
    public LiveData<List<Word>> mAllWords;

    //constructor
    public WordViewModel(Application application){
        super(application);
        context = application;
        mRepo = new WordRepo(application);
        mAllWords = mRepo.getAllWords();
        wordList = mRepo.getWordList();
        currentSightWord = new MutableLiveData<>();
        setCurrentSightWord();
    }

    //method that gets a boolean to check if th user's speech was correct
    //if the speech is correct, the nextWord()method is called
    public void checkSpeechIsCorrect(boolean isCorrect){
        if (isCorrect){
            nextWord();
        }
    }

    public void setWordList(List<Word>  words){
        wordList = words;
    }

    //does what it says
    private void setCurrentSightWord(){
        currentSightWord.setValue(wordList.get(currentIndex));
    }

    public void getSpeech(){
        SpeechHelper.Companion.getSpeechInput(context);
    }

    //increments the currentIndex and changes the currentWord MutableLive object
    public void nextWord(){
        if(currentIndex == wordList.size() - 1){
            currentIndex = 0;
        }else {
            currentIndex++;
        }

        setCurrentSightWord();
    }

}

File: app/src/main/java/com/matthewcannefax/learnsightwords/word/WordViewModel.kt
package com.matthewcannefax.learnsightwords.word

import android.app.Application
import android.content.Context
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.MutableLiveData
import com.matthewcannefax.learnsightwords.speech.SpeechHelper

class WordViewModel(app: android.app.Application) : AndroidViewModel(app) {
    val currentSightWord = MutableLiveData<Word>()
    var wordList = arrayListOf<Word>()
    var currentIndex = 0
    val context = app

    init {
        wordList= WordHelper.getWordList()
        currentSightWord.value = wordList.get(0)
    }

    fun getSpeech(context: Context){
        SpeechHelper.getSpeechInput(context)
    }

    fun nextWord(){
        if(currentIndex == wordList.size - 1){
            currentIndex = 0
        }else{
            currentIndex++
        }

        currentSightWord.value = wordList.get(currentIndex)
    }


}